### PR TITLE
[GraphBolt][CUDA] Pipelined sampling accuracy fix

### PR DESCRIPTION
**Pull request by mfbalin**
## Description
There was a bug related to the permutation of the vertices and recording the inputs to the graph fetching datapipe. This PR fixes these issues.

Now, enabling graph fetch in the dataloader or not enabling it gives the identical accuracies.

I would like to modify the node_classification example to add the overlap_graph_fetch option so that we can run the regression test for this case as well.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: tests/python/pytorch/graphbolt/impl/test_neighbor_sampler.py [hunk 1]
```diff
@@ -41,8 +41,12 @@ def get_hetero_graph():
...
 @pytest.mark.parametrize("hetero", [False, True])
 @pytest.mark.parametrize("prob_name", [None, "weight", "mask"])
-def test_NeighborSampler_GraphFetch(hetero, prob_name):
-    items = torch.arange(3)
+@pytest.mark.parametrize("sorted", [False, True])
+def test_NeighborSampler_GraphFetch(hetero, prob_name, sorted):
+    if sorted:
+        items = torch.arange(3)
+    else:
+        items = torch.tensor([2, 0, 1])
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/impl/test_neighbor_sampler.py:49]: I am a little bit confused by the unit test case, does it suggest that regardless whether the items are sorted or not, the result will be the same?

**Review by frozenbugs — APPROVED** [review]

File: tests/python/pytorch/graphbolt/impl/test_neighbor_sampler.py (hunk 1 above)
Comment (mfbalin) [reply tests/python/pytorch/graphbolt/impl/test_neighbor_sampler.py:49]: Yes, however, the code path may be a little different. Sorting is required when using hetero graphs because the extracted InSubgraph needs to have the node_type_offset field. When nodes are not sorted, it is not possible to assign a value to that field.

Comment (mfbalin) [reply tests/python/pytorch/graphbolt/impl/test_neighbor_sampler.py:49]: Whether the nodes are sorted or not, the output should match with the nonpipelined version.
